configs/module.go

```diff
@@ -179,9 +180,17 @@ func (m *Module) appendFile(file *File) hcl.Diagnostics {
 
 	for _, reqd := range file.RequiredProviders {
 		var fqn addrs.Provider
-		if reqd.Source != "" {
-			// FIXME: capture errors
-			fqn, _ = addrs.ParseProviderSourceString(reqd.Source)
+		if reqd.Source.SourceStr != "" {
+			var sourceDiags tfdiags.Diagnostics
+			fqn, sourceDiags = addrs.ParseProviderSourceString(reqd.Source.SourceStr)
+			if sourceDiags.HasErrors() {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Invalid provider source string",
+					Detail:   sourceDiags[0].Description().Detail,
```

Comment:
Yes, while that particular function only ever returns a single diagnostic, this should iterate in case that ever changes. 

Totally in the wrong place to make another proposal, but I'll do it anyway ;) .. `tfdiags.Diagnostics` already knows how to expand things in `Append`, but `hcldiags` does not. We can't share the interface, because it contains `tfdiags` struct types, but maybe we want some way to unpack in the reverse if this is encountered in more places.

Reply:
Ha, whoops, that was sloppy - thanks, both of you. I'll fix that before merging

Reply:
I'm feeling indecisive about the sourceDiags handling. Here's an alternative which folds all errors into a single one - are either more correct/preferred? 

```
diags = append(diags, &hcl.Diagnostic{
    Severity: hcl.DiagError,
    Summary:  "Invalid provider source string",
    Detail:   sourceDiags.Err().Error(),
    Subject:  &reqd.Source.DeclRange,
})
```